src/mapping/Mapping.cpp

```diff
@@ -133,7 +133,9 @@ void Mapping::map(int inputDataID,
                  input()->getDimensions(), output()->getDimensions());
   PRECICE_ASSERT(getDimensions() == output()->getDimensions(),
                  getDimensions(), output()->getDimensions());
-  PRECICE_ASSERT(input()->data(inputDataID)->getDimensions() == output()->data(outputDataID)->getDimensions(),
+  PRECICE_ASSERT((input()->data(inputDataID)->getDimensions() == output()->data(outputDataID)->getDimensions()) ||
+                     (input()->data(inputDataID)->getDimensions() == output()->data(outputDataID)->getDimensions() / 3) ||
+                     (input()->data(inputDataID)->getDimensions() == output()->data(outputDataID)->getDimensions() * 3),
```

Comment:
Could you at least add a comment that describes the purpose of this along with a doxygen `@todo` so that we don't forget?

Reply:
This assertions originate from this commit: https://github.com/ezonta/precice/commit/4b9d1f9b848ad13320d5a2d9bfa79528e4e1ac46 It originally added two assertions (one for `data()->getDimensions()` and one for a direct call to `input()->getDimensions()`/`output()->getDimensions()`). Only the first survived (because I guess only the first one is valid).

I think what this assertion is trying to restrict is that mappings are only between:

- same dimensions
- 1D and 3D
- 3D and 1D

Since we do not have support for 1D meshes yet, I think this does not yet make sense. I am removing it, also to avoid any potential side-effects to the rest of the code at the moment.